internal/pkg/manifest/svc.go

```diff
@@ -41,6 +41,47 @@ type ServiceImageWithPort struct {
 	Port         uint16 `yaml:"port"`
 }
 
+// Sidecar holds configuration for all sidecar containers in a service.
+type Sidecar struct {
+	Sidecars map[string]SidecarConfig `yaml:"sidecars"`
+}
+
+// SidecarConfig represents the configurable options for setting up a sidecar container.
+type SidecarConfig struct {
+	Port      string `yaml:"port"`
+	Image     string `yaml:"image"`
+	CredParam string `yaml:"credentialParameter"`
+}
+
+func (tc Sidecar) copyAndApply(other Sidecar) Sidecar {
```

Comment:
nit: should  the receiver be just `s`?

Reply:
oh yeah good catch! I always forget this